Sources/MapboxMaps/Offline/OfflineErrors.swift

```diff
@@ -0,0 +1,110 @@
+import Foundation
+
+internal protocol CoreErrorRepresentable {
+    associatedtype CoreErrorType
+    init(coreError: CoreErrorType)
+    init(unspecifiedError: String)
+}
+
+/// Describes the reason for a tile region download request failure.
+public enum TileRegionError: LocalizedError, CoreErrorRepresentable {
```

Comment:
Why LocalizedError instead of Error?

Reply:
Same question for StylePackError

Reply:
As a convenience to provide access to the associated message, rather than using `CustomStringConvertible`